app/src/main/java/com/limelight/AppView.java

```diff
@@ -531,6 +587,68 @@ public void run() {
         });
     }
 
+    //Saves the status of the application
+    //Removes the app from the view if the "Show hidden Apps" is not checked
+    private void saveHideStatus(AppObject app)
+    {
+        List<String> appObjects = getHideStatus();
+
+        if(!app.isHidden && appObjects.contains(String.valueOf(app.app.getAppId()))){
+            appObjects.remove(String.valueOf(app.app.getAppId()));
+            if(!showAllApps)
+            {
+                appGridAdapter.addApp(app);
+            }
+            appGridAdapter.notifyDataSetChanged();
+        }
+        else if (app.isHidden && !appObjects.contains(String.valueOf(app.app.getAppId())) ){
+            appObjects.add(String.valueOf(app.app.getAppId()));
+            if(!showAllApps)
+            {
+                appGridAdapter.removeApp(app);
+
+            }
+            appGridAdapter.notifyDataSetChanged();
+        }
+
+        String joined = TextUtils.join(",", appObjects);
+
+        OutputStream cacheOut = null;
+        try {
+            cacheOut = CacheHelper.openCacheFileForOutput(getCacheDir(), "hiddenappslist", uuidString);
```

Comment:
This probably belongs in a database but since we don't have that for apps right now, I would be okay if you put the stuff in a SharedPreferences object named `"hiddenapps-"+uuidString`. You could store the hidden state as a boolean with the key being the app ID.